dvc/utils/serialize/_yaml.py

```diff
@@ -50,6 +51,8 @@ def _get_yaml():
     # tell Dumper to represent OrderedDict as normal dict
     yaml_repr_cls = yaml.Representer
     yaml_repr_cls.add_representer(OrderedDict, yaml_repr_cls.represent_dict)
+    yaml_repr_cls.add_representer(benedict, yaml_repr_cls.represent_dict)
```

Comment:
benedict should never escape when we merge params from `--set-params`. This may result in loss of commits. 
A fix might be to merge the params in the file recursively from the `--set-params` benedict dictionary.

Reply:
Not sure I'm fully understanding, what do you mean by `escape` ?

Reply:
Removed that and replaced with just a regression test currently checking that a exception is being raised. I guess that if we go with https://github.com/iterative/dvc/issues/6423#issuecomment-908028959 we should make that test pass

Reply:
#6476 is more urgent than #6423, though we could do it together too.

Reply:
I'm not sure but maybe we could add this tests for now and discuss in https://github.com/iterative/dvc/pull/6521 how we want to handle #5477 (as it would make #6476 no longer a bug but a different exception)